### PR TITLE
prevent autopublish

### PR DESCRIPTION
**Pull request by JosuaKrause**
- [x] add more instructions about the migration
- [x] fix issues with loading global settings
- [x] fix issues with insert returning
- [x] fix autopublish

this pr solves two bugs. one is that when inserting with a `on conflict do nothing` clause the query returns no rows even with the `returning` keyword so we need to run a `select` query in that case afterwards. the other bug is that pinboards got automatically published when adding a published document. this is prevented by restricting the maximum automatic status to 1.

furthermore, the migration for pinboards is updated to allow duplicate pinboards initially. those duplicates need to be manually fixed (e.g., by renaming pinboards after the migration) and then migration can be finished properly. instructions on how to do this were added to the readme

### CODE REVIEW

File: routes/index.js [hunk 1]
```diff
@@ -815,6 +815,8 @@ exports.api.datasources = (req, res) => {
 		const { uuid } = req.session || {}
 		const { tag } = req.body || {}
 
+		// FIXME if the entry already exists we will return 500 here (because of 'one')
```
Comment (JosuaKrause) [inline routes/index.js:818]: I noticed that while fixing an issue on my side. Not sure if the current behavior here is desired. @myjyby let me know if not -- I can quickly fix it in this pr, too.

File: config/edit/index.js [hunk 2]
```diff
@@ -7,6 +7,12 @@ if (app_id === 'ap') {
 	app_obj = {...app_obj, ...require('./exp.js')};
 } else if (app_id === 'sm') {
 	app_obj = {...app_obj, ...require('./sm.js')};
+} else if (app_id === 'global') {
```
Comment (JosuaKrause) [inline config/edit/index.js:10]: this is only used for migrations

**Review by myjyby — DISMISSED** [review]
collectively reviewed and approved

**Review by myjyby — APPROVED** [review]
ok

**Pull request merged**